src/editor/EditorSplitter.svelte

```diff
@@ -0,0 +1,171 @@
+<script>
+	import * as yootils from 'yootils';
+	import { createEventDispatcher } from 'svelte';
+
+	const dispatch = createEventDispatcher();
+
+	export let type = 'columns';
+	export let pos = 50;
+	export let fixed = false;
+	export let buffer = 0;
+	export let min;
+	export let max;
+
+	let w;
+	let h;
+	$: size = type === 'vertical' ? h : w;
+
+	$: min = 100 * (buffer / size);
+	$: max = 100 - min;
+	$: pos = yootils.clamp(pos, min, max);
+
+	const refs = {};
+
+	let dragging = false;
+
+	function setPos(event) {
+		const { top, left } = refs.container.getBoundingClientRect();
+
+		const px = type === 'vertical'
+			? (event.clientY - top)
+			: (event.clientX - left);
+
+		pos = 100 * px / size;
+		dispatch('change');
+	}
+
+	function drag(node, callback) {
+		const mousedown = event => {
+			if (event.which !== 1) return;
+
+			event.preventDefault();
+
+			dragging = true;
+
+			const onmouseup = () => {
+				dragging = false;
+
+				window.removeEventListener('mousemove', callback, false);
+				window.removeEventListener('mouseup', onmouseup, false);
+			};
+
+			window.addEventListener('mousemove', callback, false);
+			window.addEventListener('mouseup', onmouseup, false);
+		}
+
+		node.addEventListener('mousedown', mousedown, false);
+
+		return {
+			destroy() {
+				node.removeEventListener('mousedown', onmousedown, false);
+			}
+		};
+	}
+
+	$: side = type === 'horizontal' ? 'left' : 'top';
+	$: dimension = type === 'horizontal' ? 'width' : 'height';
+</script>
+
+<style>
+	.container {
+		position: relative;
+		width: 100%;
+		height: 100%;
+	}
+
+	.pane {
+		position: relative;
+		float: left;
+		width: 100%;
+		height: 100%;
+		overflow: auto;
+	}
+
+	.mousecatcher {
+		position: absolute;
+		left: 0;
+		top: 0;
+		width: 100%;
+		height: 100%;
+		background: rgba(0, 0, 0, 0.0);
+	}
+
+	.divider {
+		position: absolute;
+		z-index: 10;
+		background-color: var(--surface);
+	}
+
+	.divider:hover, .divider:active{
+		filter: brightness(1.5);
+	}
+
+	.divider::after {
+		content: '';
+		position: absolute;
+	}
+
+	.horizontal {
+		padding: 0 2px;
+		width: 0;
+		height: 100%;
+		cursor: ew-resize;
+		margin-right: 50px;
+	}
+
+	.horizontal::after {
+		left: 0px;
+		top: 0;
+		width: 0px;
+		height: 100%;
+	}
+
+	.vertical {
+		padding: 2px 0;
+		width: 100%;
+		height: 0;
+		cursor: ns-resize;
+	}
+
+	.vertical::after {
+		top: 2px;
+		left: 0;
+		width: 100%;
+		height: 1px;
+	}
+
+	.left, .right, .divider {
+		display: block;
+	}
+
+	.left, .right {
+		height: 100%;
+		float: left;
+	}
+
+	.top, .bottom {
+		position: absolute;
+		width: 100%;
+	}
+
+	.top { top: 0; }
+	.bottom { bottom: 0; }
+</style>
+
+<div class="container" bind:this={refs.container} bind:clientWidth={w} bind:clientHeight={h}>
+	<div class="pane" style="{dimension}: {pos}%;">
+		<slot name="a"></slot>
```

Comment:
It's probably better to be a bit more descriptive on the slot name, so maybe do `left` and `right`.
```suggestion
		<slot name="left"></slot>
```

Reply:
I noticed there is a vertical split mode, so left and right may not be good; a and b are prolly best.